### PR TITLE
Changes to the templates to include StatefulSet for PostgreSQL and separated backend as well

### PR DESCRIPTION
**Pull request by amadhusu**
Changes to the templates to include StatefulSet for PostgreSQL and separated back-end as well

Signed-off-by: Achyut Madhusudan <amadhusu@redhat.com>

### CODE REVIEW

File: openrota/openrota-database-template.yaml [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+                valueFrom:
+                  secretKeyRef:
+                    key: pgdata
+                    name: ${DATABASE_SERVICE_NAME}
+              - name: POD_IP
+                valueFrom:
+                  fieldRef:
+                    apiVersion: v1
+                    fieldPath: status.podIP
+            image: ' '
```
Comment (amadhusu) [inline openrota/openrota-database-template.yaml]: @riprasad left this blank from the original template. When I deployed I mentioned the particular image i was trying to deploy on my cluster.

File: openrota/openrota-database-template.yaml [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+            imagePullPolicy: IfNotPresent
+            livenessProbe:
+              exec:
+                command:
```
Comment (amadhusu) [inline openrota/openrota-database-template.yaml]: @riprasad Changed the livenessProbe and readinessProbe command w.r.t testing on my OCP 4.10 cluster. We can revert this if you face problems.

File: openrota/openrota-database-template.yaml [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    sessionAffinity: None
+    type: ClusterIP
+  status:
+    loadBalancer: {}
+
+- apiVersion: v1
+  kind: PersistentVolumeClaim
+  metadata:
+    annotations:
+      kubernetes.io/reclaimPolicy: Delete
```
Comment (riprasad) [inline openrota/openrota-database-template.yaml]: reclaimPolicy should not be delete. Please refer https://source.redhat.com/groups/public/itpaas/managedplatformplushub/mppluswiki/storage_offerings

File: openrota/openrota-database-template.yaml (hunk 2 above)
Comment (riprasad) [reply openrota/openrota-database-template.yaml]: not sure what ocp version we are on MP+. We should use the preprod cluster and test the compatibility.

**Review by riprasad — CHANGES_REQUESTED** [review]
Let's not use reclaim policy as delete and use appropriate configs to avoid data loss. We already have the manifest files for the backend application. I would recommend to deploy this on the pre-prod cluster and integrate the backend.

File: openrota/openrota-database-template.yaml (hunk 3 above)
Comment (amadhusu) [reply openrota/openrota-database-template.yaml]: yeah...lets go ahead with "Retain" considering reclaimPolicy is required -> "An annotation reclaimPolicy of either “Delete” or “Retain” is required and failure to include will lead to the storage webhook to deny your request."

File: openrota/openrota-database-template.yaml (hunk 2 above)
Comment (amadhusu) [reply openrota/openrota-database-template.yaml]: I will deploy to preprod cluster tomorrow and let you know if we are fine with that.

**Pull request merged**